### PR TITLE
feat(eth): eth burn to nervos.

### PR DESCRIPTION
**Pull request by Daryl-L**

### CODE REVIEW

**Review by pygman — COMMENTED** [review]
viewed

File: offchain-modules/packages/app-rpc-server/src/handler/burn/eth.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { NetworkBase } from '../../types/network';
+import Burn from './burn';
+
+class Eth extends Burn {
+  async handle<T extends Required<NetworkBase>>(
+    payload: GenerateBridgeNervosToXchainBurnTxPayload,
+  ): Promise<GenerateTransactionResponse<T>> {
+    logger.info(`generateBridgeOutEtherumTransaction, payload: ${JSON.stringify(payload)}`);
+
+    this.checkCKBAddress(payload.recipient);
```
Comment (mkxbl) [inline offchain-modules/packages/app-rpc-server/src/handler/burn/eth.ts:16]: Miss `checkBurn`, ref: https://github.com/nervosnetwork/force-bridge/blob/85725f19cae4f5c36eacf8e351e85568ac56c6fa/offchain-modules/packages/app-rpc-server/src/handler.ts#L88

* check asset in tokenToAssetIdMap(AssetManager.sol)
* check amount >= minimalAmount
* check sudtSize <= ForceBridgeCore.config.ckb.sudtSize
* besides check recipientAddressFormat, better check recipientAddressLength <= dbRecipientFieldLength

File: offchain-modules/packages/app-rpc-server/src/handler/burn/eth.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    const contract = new ethers.Contract(
+      ForceBridgeCore.config.eth.assetManagerContractAddress,
+      abi,
+      new ethers.providers.JsonRpcBatchProvider(ForceBridgeCore.config.eth.rpcUrl),
```
Comment (mkxbl) [inline offchain-modules/packages/app-rpc-server/src/handler/burn/eth.ts:21]:
```suggestion
      new ethers.providers.JsonRpcProvider(ForceBridgeCore.config.eth.rpcUrl),
```

File: offchain-modules/packages/app-rpc-server/src/handler/burn/eth.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    );
+
+    const amount = ethers.utils.parseUnits(payload.amount);
```
Comment (mkxbl) [inline offchain-modules/packages/app-rpc-server/src/handler/burn/eth.ts:24]:
```suggestion
    const amount = ethers.utils.parseUnits(payload.amount, 0);
```

File: offchain-modules/packages/app-rpc-server/src/handler/burn/eth.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    const recipient = stringToUint8Array(payload.recipient);
+    const extraData = '0x';
+
+    const tx = await contract.populateTransaction.burn(payload.asset, amount, recipient, extraData);
```
Comment (mkxbl) [inline offchain-modules/packages/app-rpc-server/src/handler/burn/eth.ts:28]: Bridge fee missed:

```suggestion
    const tx = await contract.populateTransaction.burn(payload.asset, amount, recipient, extraData, {value: bridgeFee});
```

**Pull request merged**